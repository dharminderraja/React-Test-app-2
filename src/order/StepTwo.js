import React, { Component } from 'react';

class StepTwo extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      restaurant: '',
      errors: {}
    };
  }

  componentWillMount() {
    const { restaurant } = this.props.data || {};
    if (restaurant) {
      this.setState({ restaurant });
    }
  }

  onChange(property, value) {
    this.setState({ [property]: value });
  }

  validate() {
    let { errors, restaurant } = this.state;

    errors['restaurant'] = !restaurant ? 'Please select restaurant' : '';
    this.setState({ errors })

    return Object.keys(errors).every(k => !errors[k]);
  }

  nextStep() {
    const { updateData, nextTab, disableTabs } = this.props;
    const { restaurant } = this.state;

    if (this.validate()) {
      updateData({ restaurant });
      nextTab();
    } else {
      disableTabs(['tabThree', 'tabFour']);
    }
  }

  render() {
    const { restaurants, prevTab } = this.props;
    const { restaurant, errors } = this.state;

    return (
      <div className='step-detail'>
        <div>
          <span className='title'>Please select a Restaurant</span>
          <select className='meal' value={restaurant || ''} onChange={evt => this.onChange('restaurant', evt.target.value)}>
            <option value=''>...</option>
            {(restaurants || []).map((i, idx) => (<option key={idx} value={i}>{i}</option>))}
          </select>
          {errors['restaurant'] && <span className='alert-warning'>{errors['restaurant']}</span>}
        </div>
        <div className='nav-step'>
          <button className='prev' onClick={() => prevTab()}>Previous</button>
          <button className='next' onClick={() => this.nextStep()}>Next</button>
        </div>
      </div>
    );
  }
}

export default StepTwo;
