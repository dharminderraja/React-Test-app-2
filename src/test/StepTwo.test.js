import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import StepTwo from '../order/StepTwo';

describe('Step Two', () => {
  it('should be defined', () => {
    expect(StepTwo).toBeDefined();
  });

  it('should render correctly', () => {
    const component = renderer.create(<StepTwo />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show values correctly', () => {
    const tree = shallow(
      <StepTwo
        restaurants={['Mc Donalds', 'Taco Bell', 'BBQ Hut']}
        data={{ restaurant: 'Mc Donalds' }}
        prevTab={() => { }} />
    );

    expect(tree.find('.step-detail .meal').children()).toHaveLength(4);
    expect(tree.find('.step-detail .meal').getElement().props.value).toEqual('Mc Donalds');
  });
});