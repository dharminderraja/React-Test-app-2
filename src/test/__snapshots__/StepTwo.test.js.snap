// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Two should render correctly 1`] = `
<div
  className="step-detail"
>
  <div>
    <span
      className="title"
    >
      Please select a Restaurant
    </span>
    <select
      className="meal"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        ...
      </option>
    </select>
  </div>
  <div
    className="nav-step"
  >
    <button
      className="prev"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="next"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
